Translation of the "Fragment" page (#455)
reactjs/fr.react.dev: src/content/reference/react/Fragment.md

<Intro>

> `<Fragment>`, souvent utilisé avec la syntaxe `<>...</>`, vous permet de grouper des éléments sans balise englobante.

Author: Pas sur de devoir traduire "wrapper node", ni c'est si la traduction la plus adaptée

---

## Référence {/*reference*/}

### `<Fragment>` {/*fragment*/}

Englober des éléments dans un `<Fragment>` pour les grouper dans des situations ou vous avez besoin d'un seul élément. Grouper des éléments dans un `<Fragment>` n'a pas d'effet sur le DOM rendu ; c'est comme si les éléments n'étaient pas groupés. La balise vide `<></>` en JSX est utilisée la plupart du temps afin de raccourcir l'écriture de `<Fragment></Fragment>`.

#### Props {/*props*/}

> - **optionnel** `key`: Les Fragments déclarés explicitement avec la syntaxe `<Fragment>` peuvent contenir des [keys.](/learn/rendering-lists#keeping-list-items-in-order-with-key)

Reviewer: ```suggestion
- `key` **optionnel** : les fragments déclarés explicitement avec la syntaxe `<Fragment>` peuvent avoir une [keys](/learn/rendering-lists#keeping-list-items-in-order-with-key).
```

- Tu as perdu de vue l'ordre français des mots 😁 Ici l'adjectif serait en position attribut, donc après.
- Espaces insécables devant les signes de ponctuation doubles.
- Pas de majuscules en français après un deux-points.
- "have", pas "contain", comme on parle de props ici, donc "avoir".
- La VO n'est pas claire mais un élément React a au maximum une `key`.
- En anglais, les règles de ponctuation font que le plus souvent, une ponctuation à la fin d'un groupe (par exemple paire de guillemets, ou sur le web hyperlien) est *au sein* du groupe, comme dans ces docs les points finaux toujours au sein des liens. En français non, on sort donc le point pour le mettre après le lien.

#### Limitations {/*caveats*/}

> - Si vous souhaitez utiliser des `key` dans un Fragment, vous ne pouvez pas utiliser la syntaxe `<>...</>`. Vous devez explicitement importer `Fragment` depuis `'react'` et rendre `<Fragment key={yourKey}>...</Fragment>`.

Reviewer: ```suggestion
- Si vous souhaitez utiliser une `key` dans un Fragment, vous ne pouvez pas utiliser la syntaxe `<>...</>`. Vous devez explicitement importer `Fragment` depuis `'react'` et écrire `<Fragment key={yourKey}>...</Fragment>`.
```

Comme l'indiquent les BP de trad, on évite au maximum d'utiliser "rendre" comme verbe. Chaque fois que le distingo n'est pas critique, on utilisera selon le contexte "afficher", "renvoyer" ou "écrire" (ici on parle de la syntaxe à employer). Si vraiment on parle de la notion technique React de *rendering*, on dira plutôt "faire le rendu" ou "faire son rendu" si le sujet est le composant qui fait le rendu.

> - React ne [réinitialise pas l'état](/learn/preserving-and-resetting-state) quand vous passez de rendre `<><Child /></>` à `[<Child />]` et inversement, ou ni si vous passez de rendre `<><Child /></>` à `<Child />` et inversement. Ça marche seulement à un niveau plus profond : par exemple, passer de `<><><Child /></></>` à `<Child />` réinitialise l'état. Consultez la sémantique précise [ici.](https://gist.github.com/clemmy/b3ef00f9507909429d8aa0d3ee4f986b)

Reviewer: ```suggestion
- React ne [réinitialise pas l'état](/learn/preserving-and-resetting-state) quand vous passez d'une structure `<><Child /></>` à `[<Child />]` et inversement, ni quand vous passez de `<><Child /></>` à `<Child />` et inversement. Ça ne marche qu’à un niveau de profondeur : par exemple, passer de `<><><Child /></></>` à `<Child />` réinitialise l'état. Consultez [la sémantique précise ici](https://gist.github.com/clemmy/b3ef00f9507909429d8aa0d3ee4f986b).
```

- Quelques écarts de sens ("plus profond")
- "Rendre", etc.
- On évite d'utiliser des subordonnées basées sur un infinitif, on va simplifier autant que possible.
- Amélioration de l'accessibilité du lien en évitant juste "ici" comme texte (souci de la VO aussi).

---

## Utilisation {/*usage*/}

### Renvoyer plusieurs éléments {/*returning-multiple-elements*/}

> Utilisez `Fragment`, ou la syntaxe équivalente `<>...</>`, afin de grouper plusieurs éléments ensemble. Vous pouvez l'utiliser pour mettre plusieurs éléments à la place d'un élément. Par exemple, un composant ne peut renvoyer qu'un élément unique, mais avec un Fragment vous pouvez grouper plusieurs éléments et renvoyer ce groupe :

Reviewer: ```suggestion
Utilisez `Fragment`, ou la syntaxe équivalente `<>...</>`, afin de grouper plusieurs éléments ensemble. Vous pouvez l'utiliser pour passer plusieurs éléments là où un seul est attendu. Par exemple, un composant ne peut renvoyer qu'un seul élément, mais avec un Fragment vous pouvez grouper plusieurs éléments et renvoyer ce groupe :
```

- Clarification du cas d'usage

}
```

> Les Fragments sont utiles car grouper des éléments avec un Fragment n'as pas d'effet sur la mise en page ou les styles, contrairement à l'utilisation d'un autre balise englobante du DOM. Si vous inspectez cet exemple avec les outils du navigateur, vous verrez que toutes les balises `<h1>` et `<p>` du DOM apparaissent en tant frères et sœur sans balise englobante :

Author: Pas sur de la traduction de `siblings` => `enfants` ?

Reviewer: En effet, ce n'est pas ça ("children"). "Siblings" n'a hélas pas de traduction satisfaisante en un mot, ça désigne les frères et sœurs IRL, donc les éléments de même parent en fait.

> Les Fragments sont utiles car grouper des éléments avec un Fragment n'as pas d'effet sur la mise en page ou les styles, contrairement à l'utilisation d'un autre balise englobante du DOM. Si vous inspectez cet exemple avec les outils du navigateur, vous verrez que toutes les balises `<h1>` et `<p>` du DOM apparaissent en tant frères et sœur sans balise englobante :

Reviewer: ```suggestion
Les Fragments sont utiles car grouper des éléments avec un Fragment n'as pas d'effet sur la mise en page ou les styles, contrairement à l'enrobage par une balise conteneur telle qu’un élément DOM. Si vous inspectez cet exemple avec les outils du navigateur, vous verrez que toutes les balises `<h1>` et `<p>` du DOM apparaissent au même niveau (avec le même parent), sans balise enrobante :
```

<Sandpack>

```js
export default function Blog() {
  return (
    <>

>       <Post title="Une mise à jour" body="Ça fait un moment que je n'ai pas posté..." />

Reviewer: ```suggestion
      <Post title="Des nouvelles" body="Ça fait un moment que je n’ai pas écrit..." />
```

Soyons le plus idiomatiques possible vu le contexte sémantique 😄

<DeepDive>

#### Comment utiliser un Fragment sans la syntaxe spéciale ? {/*how-to-write-a-fragment-without-the-special-syntax*/}

> L'exemple ci-dessus est equivalent à importer `Fragment` depuis React :

Reviewer: ```suggestion
L'exemple ci-dessus revient à importer `Fragment` depuis React :
```

"est équivalent à" + infinitif, c'est ampoulé.

}
```

> Normalement vous n'aurez pas besoin d'utiliser ça à moins d'avoir besoin de [donner des `key` à votre `Fragment`.](#rendering-a-list-of-fragments)

Reviewer: ```suggestion
Normalement vous n'aurez pas besoin d'utiliser ça à moins de vouloir [associer une `key` à votre `Fragment`](#rendering-a-list-of-fragments).
```

- Double "avoir besoin" reformulé
- Point hors du lien

</DeepDive>

---

> ### Initialiser plusieurs éléments à une variable {/*assigning-multiple-elements-to-a-variable*/}

Reviewer: ```suggestion
### Affecter plusieurs éléments à une variable {/*assigning-multiple-elements-to-a-variable*/}
```

"Assign" ne signifie pas "Initialiser" ?!

> Comme pour tous les autres éléments, vous pouvez initialiser un Fragment à des variables, les passer en tant que props, etc :

Reviewer: ```suggestion
Comme pour tous les autres éléments, vous pouvez affecter des Fragment à des variables, les passer en tant que props, etc. :
```

  );
  return (
    <AlertDialog buttons={buttons}>

>       Êtes vous sûr de vouloir quittez cette page ?

Reviewer: ```suggestion
      Êtes vous sûr·e de vouloir quitter cette page ?
```

    </AlertDialog>
  );
}
```

---

> ### Englober des éléments avec du texte {/*grouping-elements-with-text*/}

Reviewer: ```suggestion
### Grouper des éléments et du texte {/*grouping-elements-with-text*/}
```

Écart de sens.

> Vous pouvez utilisez `Fragment` pour grouper des textes avec des éléments :

Reviewer: ```suggestion
Vous pouvez utiliser `Fragment` pour grouper du texte et des éléments :
```

Un peu de fatigue sur les accords et infinitifs 😉

---

> ### Rendre une liste de Fragments {/*rendering-a-list-of-fragments*/}

Reviewer: ```suggestion
### Afficher une liste de Fragments {/*rendering-a-list-of-fragments*/}
```

La distinction n'est pas importante ici, on peut dire "afficher".

> Voici une situation ou vous avez besoin d'écrire explicitement `Fragment` plutôt que d'utiliser la syntaxe `<></>`. Quand vous [rendez plusieurs éléments dans une boucle](/learn/rendering-lists), vous avez besoin d'assigner une `key` à chaque élément. Si les éléments compris dans une boucle sont des Fragments, vous devez utiliser la syntaxe classique JSX afin de fournir l'attribut `key` :

Reviewer: ```suggestion
Voici un cas de figure où vous devez écrire `Fragment` explicitement plutôt que d'utiliser la syntaxe `<></>`. Quand vous [affichez plusieurs éléments dans une boucle](/learn/rendering-lists), vous avez besoin d'associer une `key` à chaque élément. Si les éléments compris dans la boucle sont des Fragments, vous devez utiliser la syntaxe classique JSX afin de fournir la prop `key` :
```

La VO n'est pas parfaite… Elle utilise "the" au lieu de "some" (le truc reste vrai pour tout fragment dans la liste, même si le reste est constitué d'éléments classique), et elle entretient la confusion (pas mal présente dans les nouvelles docs) entre "prop" et "attribut". Je tends à vouloir préserver le distingo historique qui réserve "attribut" aux éléments natifs et "prop" à ce qui touche à React, **surtout** pour les props techniques que sont `key`, `ref` et `children`.

}
```

> Vous pouvez inspecter le DOM pour vérifier qu'il n'y a pas de balise englobante autour des enfants du Fragment :

Reviewer: ```suggestion
Vous pouvez inspecter le DOM pour vérifier qu'il n'y a pas d'élément enrobant autour des enfants du Fragment :
```

"Balise" est un terme de syntaxe. Le DOM ne contient pas de balise, que des nœuds (essentiellement texte et éléments). `#ayatollah`

<Sandpack>

```js
import { Fragment } from 'react';

const posts = [

>   { id: 1, title: 'Une mise à jour', body: "Ça fait un moment que je n'ai pas posté..." },

Reviewer: ```suggestion
  { id: 1, title: 'Des nouvelles', body: "Ça fait un moment que je n'ai pas écrit..." },
```

Cohérence avec la correction d'avant.
